docs/adr/0008_signing_verification.md

```diff
@@ -0,0 +1,247 @@
+# Next Generation Component Constructor Support
+
+* **Status**: proposed
+* **Deciders**: OCM Technical Steering Committee
+* **Date**: 2025.08.13
+
+**Technical Story**:
+Enable easy interpretation of component descriptors in normalized form and provide a consistent experience for signing and verifying component descriptors.
+
+---
+
+## Context and Problem Statement
+
+When users want to verify the integrity of a component version, they can run:
+
+```shell
+ocm verify componentversion --signature mysig --verifier mypublickey ghcr.io/open-component-model/ocm//ocm.software/ocm:0.17.0
+```
+
+This will:
+
+1. Download the component version descriptor from the repository.
+2. Inspect the [`signatures`](https://github.com/open-component-model/ocm-spec/blob/main/doc/01-model/03-elements-sub.md#signatures) field:
+
+   ```yaml
+   signatures:
+     - name: mysig
+       digest:
+         hashAlgorithm: sha256
+         normalisationAlgorithm: jsonNormalisation/v1
+         value: cf08abae08bb874597630bc0573d941b1becc92b4916cbe3bef9aa0e89aec3f6
+       signature:
+         algorithm: RSASSA-PKCS1-V1_5
+         mediaType: application/vnd.ocm.signature.rsa
+         value: 390157b7...75ab2705d6
+   ```
+3. Verify the signature using the configured verifier from `.ocmconfig`.
+
+Signing uses the analogous command:
+
+```shell
+ocm sign componentversion --signature mysig --signer myprivatekey ghcr.io/open-component-model/ocm//ocm.software/ocm:0.17.0
+```
+
+This downloads, normalizes, signs, and re-uploads the descriptor. Signing and verification configs live in `.ocmconfig`.
+
+---
+
+## Decision Drivers
+
+* **Simplicity**: Keep signing and verification decoupled from normalization internals.
+* **Extensibility**: Support new algorithms and key types via plugins.
+* **Maintainability**: Clear contracts and separation of concerns to enable testing.
+
+---
+
+## Outcome
+
+Implement a plugin-driven transformation and signing system based on a **ComponentSignatureHandler** contract to compute and interpret normalized digests and to sign and verify these digests.
```

Comment:
`Implement a plugin-driven signing/verification system based on a **ComponentSignatureHandler** contract to compute normalized digests and sign/verify them.
`